sockeye/utils.py

```diff
@@ -271,49 +272,99 @@ def get_num_gpus() -> int:
 
 
 @contextmanager
-def acquire_gpu(lock_dir: str = "/var/lock", retry_wait: int = 10):
+def acquire_gpus(num_gpus:int, lock_dir: str = "/tmp", retry_wait_min: int = 30, retry_wait_rand: int = 30):
```

Comment:
but then also the @contextmanager makes the whole function a context manager. So technically calling the function gives you a context manager object. So overall I'm not sure how to annotate the function and I'd rather just leave out the annotation of the return type.